wlauto/tests/test_environment.py

```diff
@@ -0,0 +1,289 @@
+#    Copyright 2013-2015 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+# pylint: disable=W0231,W0613,E0611,W0603,R0201
+from unittest import TestCase
+
+from nose.tools import assert_equal, assert_raises
+
+from wlauto.utils.environment_manager import (init_environment, reset_environment,
+                                              activate_environment, once,
+                                              once_per_class, once_per_instance)
+
+class TestClass(object):
+
+    def __init__(self):
+        self.count = 0
+
+    @once
+    def initilise_once(self):
+        self.count += 1
+
+    @once_per_class
+    def initilise_once_per_class(self):
+        self.count += 1
+
+    @once_per_instance
+    def initilise_once_per_instance(self):
+        self.count += 1
+
+
+class SubClass(TestClass):
+
+    def __init__(self):
+        super(SubClass, self).__init__()
+
+    @once
+    def initilise_once(self):
+        self.count += 1
+
+    @once_per_class
+    def initilise_once_per_class(self):
+        self.count += 1
+
+    @once_per_instance
+    def initilise_once_per_instance(self):
+        self.count += 1
+
+
+class SubSubClass(SubClass):
```

Comment:
`SubClass` and `SubSubClass` are overriding all of the methods of the `TestClass`, which kinda defeats the point of testing the inheritance hierarchy. You want to subclass without re-implementing, so that derived instances will call the base methods.